src/platform_util.cpp

```diff
@@ -74,7 +74,7 @@ namespace libtorrent {
 			if (rl.rlim_cur == rlim_infinity)
 				return std::numeric_limits<int>::max();
 
-			return rl.rlim_cur <= std::numeric_limits<int>::max()
+			return rl.rlim_cur <= std::numeric_limits<unsigned long long>::max()
```

Comment:
`rlim_t` is not guaranteed to be an `unsigned long long`.

also, this changes the semantics of the code, to be incorrect. The reason this is compared against the largest value an `int` can hold is because this function returns an `int`. If anything, you can *cast* the result to `rlim_t`.

The warning you get on this line is a false positive, and I think only old compilers warn on this. The right hand side in this comparison is a *constant*. The compiler can tell that this constant fits in `rlim_t` without losing precision.

Reply:
The compiler version is gcc 9.3.0, and here's the warning:

```
/home/zywo/libtorrent/src/platform_util.cpp: In function ‘int libtorrent::max_open_files()’:
/home/zywo/libtorrent/src/platform_util.cpp:77:23: warning: comparison of integer expressions of different signedness: ‘rlim_t’ {aka ‘long unsigned int’} and ‘int’ [-Wsign-compare]
   77 |    return rl.rlim_cur <= std::numeric_limits<int>::max()
      |           ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
ok, so it's a false positive in GCC then. I assume GCC wouldn't warn if the max constant was changed to `INT_MAX`. This patch still breaks the code though.

Reply:
By using `INT_MAX` and including `climits` header, there's no warning.

If the change is useless, we can close this PR.

Reply:
the *current* patch is wrong. But you could force push to it to make it compare against the correct value again, and also fix the warning